### PR TITLE
Create app-hooks.md

### PR DESCRIPTION
**Pull request by purplecabbage**
Document the use of app tooling lifecycle event hooks.

### CODE REVIEW

**Comment by tmathern** [comment]
For the sake of having a complete documentation, could you add a diagram or "something" that shows the aio app lifecycle and when hooks are called? For instance, for deploy:
`pre-app-deploy -> deploy -> post-app-deploy` ? So there is the full lifecycle in only one documentation.

Also, do you support something like pre-pack and post-pack? pre-build and post-build?

File: guides/app-hooks.md [hunk 1]
```diff
@@ -0,0 +1,28 @@
+
+# Firefly application tooling lifecycle event hooks
```
Comment (meryllblanchet) [inline guides/app-hooks.md:2]: Replace Firefly with Project Firefly

File: guides/app-hooks.md [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+
+Firefly applications created with our tooling are inherently npm packages.
```
Comment (meryllblanchet) [inline guides/app-hooks.md:4]: Replace Firefly with Project Firefly

File: guides/app-hooks.md [hunk 3]
```diff
@@ -0,0 +1,28 @@
...
+    "pre-app-deploy": "echo pre-app-deploy",
+    "post-app-deploy": "echo post-app-deploy"
+  }
+  ```
+
+![aio-app-deploy lifecycle](../images/aio-app-deploy.png)
+
+# NPM script hooks
+
+Use of Firefly event hooks does not interfere with use of npm scripts, however you will need to use `npm run ..` to trigger them.
```
Comment (meryllblanchet) [inline guides/app-hooks.md:24]: Replace Firefly with Project Firefly

**Comment by meryllblanchet** [comment]
@purplecabbage @sarahxxu this could fit into a bigger `Contribution` guide.
WDYT?

**Pull request merged**